pulp_service/pulp_service/app/settings.py

```diff
@@ -4,3 +4,7 @@
 .. _Plugin Writer's Guide:
     https://docs.pulpproject.org/pulpcore/plugins/plugin-writer/index.html
 """
+
+RH_IDENTITY_HEADER = "HTTP_X_RH_IDENTITY"
+SUBSCRIPTION_API_URL = "https://subscription.stage.api.redhat.com/svcrest/subscription/v5/featureStatus?accountId=5894300&features=RHEL-OS-x86_64&features=RHEL-EUS-x86_64"
+SUBSCRIPTION_CERT_PATH = "/var/certs/pulp_nonprod_cert.pem"
```

Comment:
I think the key (and probably the cert itself too) to be in the vault and this should read from the vault. I'm not 100% sure how that works, but I think that's the idea. Also we'll have one cert/key for staging and a different one for prod.

Reply:
There's a `volumeMount` on the deployment file to deal with that. Sadly the `requests` lib can't read a cert from an envvar, so I had to mount the certificate as a file in the container and then point `requests` to use that cert.

Reply:
Mmm yeah I see the `volumeMount`, but where does the .pem file actually live?

Also a few years ago we tried to move away from requests as our library in favor of aiohttp.client. For example, AIUI we don't actually declare a dependency on requests (at least pulpcore doesn't). Here's an example of a client request using in memory certs with aiohttp.client.

```
import ssl
import aiohttp
import asyncio

# In-memory certificate and key (these should be in PEM format)
cert_data = b"""-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----"""
key_data = b"""-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----"""

# Create an SSL context
ssl_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
ssl_context.load_cert_chain(certfile=cert_data, keyfile=key_data)

# Function to make a request
async def fetch_with_cert():
    async with aiohttp.ClientSession() as session:
        async with session.get("https://example.com", ssl=ssl_context) as response:
            print(await response.text())

# Run the request in an event loop
asyncio.run(fetch_with_cert())
```

What do you think about all this?